google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -143,18 +144,26 @@ public Project getProject() {
           deploy, isFlexCompat));
     } else if (targetEnvironment.isFlexible()) {
       try {
-        // Checks if the Yaml or Dockerfile exist.
-        Optional<FlexibleRuntime> runtimeOptional =
-            AppEngineProjectService.getInstance().getFlexibleRuntimeFromAppYaml(
-                deploymentConfiguration.getAppYamlPath());
+        AppEngineFlexibleFacet flexFacet =
+            AppEngineFlexibleFacet.getFacetByModuleName(
+                deploymentConfiguration.getModuleName(), project);
 
-        if (!Files.exists(Paths.get(deploymentConfiguration.getAppYamlPath()))) {
+        if (flexFacet == null
+            || !Files.exists(Paths.get(flexFacet.getConfiguration().getAppYamlPath()))) {
```

Comment:
Interesting..I just testing the behavior for point 1:

After deleting the module:
- Attempting to just rerun the deploy config gives the same error as printed above "..specified app.yaml ..does not exist ...". This is just an error message in the deploy console - no IDE exception.
- Editing the run config that previously was set to the now deleted module, has the module automatically unset:

![image](https://user-images.githubusercontent.com/1735744/26987240-c108a5c4-4d18-11e7-93cb-1ef280fffb8a.png)

I think, again, that these are reasonable behaviors. But we may be able to improve error messaging as you suggest in your other comment. wdyt?